engine/runtime-parser/src/test/java/org/enso/compiler/core/EnsoParserMultiThreadedTest.java

```diff
@@ -0,0 +1,38 @@
+package org.enso.compiler.core;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ForkJoinPool;
+import org.enso.compiler.core.ir.Module;
+import org.junit.Test;
+
+public class EnsoParserMultiThreadedTest {
+  @Test
+  public void stressLoadFromManyThreads() throws Exception {
+    List<Callable<Module>> cases = new ArrayList<>();
+    final var testCases = 1000;
+    for (var i = 0; i < 2 * testCases; i++) {
+      var number = i % testCases;
+      var code =
+          """
+            from Standard.Base import all
+            main = %n
+            """
+              .replace("%n", "" + number);
+      cases.add(
+          () -> {
+            return EnsoParser.compile(code);
+          });
+    }
+
+    var results = ForkJoinPool.commonPool().invokeAll(cases);
```

Comment:
This is a port of a test introduced by
- #11174

into this branch. It just replaces the `EnsoParser` instance with a static method call.